### PR TITLE
YLP-1092 Add HCP profession in yourloops HCP signup form and account preferences

### PR DESCRIPTION
**Pull request by yanngit**

### CODE REVIEW

File: locales/en/yourloops.json [hunk 1]
```diff
@@ -83,6 +83,8 @@
   "current-password": "Current password",
   "date-picker-header-date-format": "MMMM YYYY",
   "date-picker-toolbar-date-format": "ddd, MMM D",
+  "diabeto": "Diabetologist",
```
Comment (yvon-dblg) [inline locales/en/yourloops.json:86]: I would have prefix the keys with some like "hcp-job-" for example
So we have some context in crowdin for the translations.

Comment (yanngit) [reply locales/en/yourloops.json]: Done

File: packages/yourloops/components/routes.tsx [hunk 2]
```diff
@@ -95,12 +96,15 @@ export const PrivateRoute = (props: RouteProps): JSX.Element | null => {
   } else if (!renewConsentPath && user.shouldRenewConsent()) {
     component = <Redirect to={{ pathname: "/renew-consent", state: { from: props.location } }} />;
 
+    // } else if (!renewConsentPath && !hcpPreferencesPath && user.role === UserRoles.hcp && user.shouldUpdateHcpProfession()) {
```
Comment (fatb38) [inline packages/yourloops/components/routes.tsx:99]: What is this commented code for ?

File: packages/yourloops/pages/profile/profile.tsx [hunk 3]
```diff
@@ -408,6 +415,28 @@ const ProfilePage = (props: ProfilePageProps): JSX.Element => {
...
+                  value={hcpProfession}
+                  error={errors.hcpProfession}
+                  onChange={createHandleSelectChange(setHcpProfession)}>
+                  {HcpProfessionList.filter((item) => item !== HcpProfession.empty).map((item) => (
+                    <MenuItem id={`profile-hcp-profession-menuitem-${item}`} key={item} value={item}>
+                      {t(item)}
+                    </MenuItem>
+                  ))}
+                </Select>
+                { errors.hcpProfession &&
```
Comment (fatb38) [inline packages/yourloops/pages/profile/profile.tsx:434]:
```suggestion
                {errors.hcpProfession &&
```

File: packages/yourloops/pages/profile/profile.tsx [hunk 4]
```diff
@@ -408,6 +415,28 @@ const ProfilePage = (props: ProfilePageProps): JSX.Element => {
...
+                  <FormHelperText error={true}>{t("profile-hcp-profession-helper-text")}</FormHelperText>
```
Comment (fatb38) [inline packages/yourloops/pages/profile/profile.tsx:435]: can be simplified
```suggestion
                  <FormHelperText error>{t("profile-hcp-profession-helper-text")}</FormHelperText>
```

File: packages/yourloops/pages/signup/signup-profile-form.tsx [hunk 5]
```diff
@@ -210,6 +221,34 @@ function SignUpProfileForm(props: SignUpFormProps): JSX.Element {
...
+        >
+          <InputLabel id="hcp-profession-selector-input-label">
+            {t("hcp-profession-input-label")}
+          </InputLabel>
+          <Select
+            labelId="hcp-profession-selector-label"
+            label={t("signup-hcp-profession")}
+            id="hcp-profession-selector"
+            value={state.formValues?.hcpProfession}
+            onBlur={() => validateHcpProfession()}
```
Comment (fatb38) [inline packages/yourloops/pages/signup/signup-profile-form.tsx:240]: can be simplified
```suggestion
            onBlur={validateHcpProfession}
```

File: packages/yourloops/test/pages/stepper/signup-profile-form.test.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    if (state.formValues.accountRole !== "hcp") {
+      dispatch({
+        type: "EDIT_FORMVALUE",
+        key: "accountRole",
+        value: "hcp",
+      });
+    }
+  }, [dispatch, state.formValues.accountRole]);
+
+  return (null);
```
Comment (fatb38) [inline packages/yourloops/test/pages/stepper/signup-profile-form.test.tsx:51]: can be simplified
```suggestion
  return null;
```

File: packages/yourloops/test/pages/stepper/signup-profile-form.test.tsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+function TestSignupProfileForm(): void {
+  let container: HTMLElement | null = null;
+
+  const mockHandleBack = () => {
+    return true;
+  };
+  const mockHandleNext = mockHandleBack;
```
Comment (fatb38) [inline packages/yourloops/test/pages/stepper/signup-profile-form.test.tsx:60]: Why are these functions here ? fill in SignUpProfileForm props ?

**Review by fatb38 — CHANGES_REQUESTED** [review]

File: packages/yourloops/test/pages/stepper/signup-profile-form.test.tsx (hunk 7 above)
Comment (yanngit) [reply packages/yourloops/test/pages/stepper/signup-profile-form.test.tsx]: just to mock but i used _.noop instead

**Review by fatb38 — COMMENTED** [review]
LGTM

**Review by fatb38 — APPROVED** [review]

**Pull request merged**